woocommerce/wc-admin
# Hook up leaderboards endpoint to dashboard

Fixes #1972 

Hooks up the dashboard leaderboards to the new leaderboards endpoint and allows custom leaderboards to be added via the endpoint hook.

### Screenshots

<img width="637" alt="Screen Shot 2019-04-09 at 4 11 16 PM" src="https://user-images.githubusercontent.com/10561050/55783977-2c120d00-5ae2-11e9-86a1-16b2170b8879.png">


### Detailed test instructions:

1. Go to the dashboard.
2. Check that leaderboards are shown with loading rows while being fetched.
3. Check that data is fetched and shown properly for leaderboards.
4. Check that leaderboards can be toggled on/off still and that this persists after refreshing.
5. Check that changing the rows option still affects the shown number of rows.
6. Check that changing the date range affects leaderboards.
7. Check that custom leaderboards show up here.

### Test Data

Here's a quick copy/paste example to test custom leaderboards:

```php
add_filter( 'woocommerce_leaderboards', function( $leaderboards ) {
	$leaderboards[] = array(
		'id'      => 'dragons',
		'label'   => __( 'Top Dragons', 'woocommerce-admin' ),
		'headers' => array(
			array(
				'label' => __( 'Name', 'woocommerce-admin' ),
			),
			array(
				'label' => __( 'Fire Color', 'woocommerce-admin' ),
			),
		),
		'rows'    => array(
			array(
				array(
					'display' => '<a href="#">Drogon</a>',
					'value' => 'Drogon',
				),
				array(
					'display' => 'Orange',
					'value' => 'orange',
				),
			),
			array(
				array(
					'display' => '<a href="#">Rhaegal</a>',
					'value' => 'Rhaegal',
				),
				array(
					'display' => 'Orange',
					'value' => 'orange',
				),
			),
			array(
				array(
					'display' => '<a href="#">Viserion</a>',
					'value' => 'Viserion',
				),
				array(
					'display' => 'Blue',
					'value' => 'blue',
				),
			),
		),
	);
	return $leaderboards;
} );
```
